File: OC/内容层动画/WGOCCoreAnimation/WGOCCoreAnimation/水纹动画/WGWaterWaveBtnViewController.m
//
//  WGWaterWaveBtnViewController.m
//  WGOCCoreAnimation
//
//  Created by wanggang on 2019/5/22.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGWaterWaveBtnViewController.h"
#import "WGWaterWaveBtn.h"

@interface WGWaterWaveBtnViewController ()

@property (nonatomic, strong) WGWaterWaveBtn *btn;

@end

@implementation WGWaterWaveBtnViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.btn = [[WGWaterWaveBtn alloc] initWithFrame:CGRectMake(20, SCREEN_HEIGHT/2, SCREEN_WIDTH-40, 70)];
    _btn.backgroundColor = UIColor.redColor;
    [_btn setTitle:@"点我" forState:UIControlStateNormal];
    [_btn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    [_btn addTarget:self action:@selector(btnClicked: event:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:self.btn];
}

- (void)btnClicked:(UIButton *)sender event:(UIEvent *)event{
    
    [self.btn startAniWithEvent:event];
}

@end

File: Swift/内容层动画/WGCoreAnimation/WGCoreAnimation/水纹动画/WGWaterWaveBtnViewController.swift
//
//  WGWaterWaveBtnViewController.swift
//  WGCoreAnimation
//
//  Created by wanggang on 2019/4/23.
//  Copyright © 2019 wanggang. All rights reserved.
//

import UIKit

class WGWaterWaveBtnViewController: UIViewController {

    var btn: WGWaterWaveBtn!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = UIColor.white
        
        btn = WGWaterWaveBtn(frame: CGRect(x: 20, y: 100, width: WgWith-20*2, height: 50))
        btn.setTitle("点我", for: .normal)
        btn.setTitleColor(UIColor.black, for: .normal)
        btn.addTarget(self, action: #selector(btnClicked(sender:event:)), for: .touchUpInside)
        view.addSubview(btn)
    }
    
    @objc func btnClicked(sender: WGWaterWaveBtn, event: UIEvent) {
        
        sender.startAnimation(event: event)
    }

}
